app/controllers/order_controller.rb

```diff
@@ -0,0 +1,145 @@
+class OrderController < ApplicationController
+  before_action :authenticate_user!
+  before_action :current_user_cart
+
+  def index
+    @orders = current_user_orders.order(created_at: :desc)
+  end
+
+  def show
+    @order = current_user_orders.find_by(id: params[:id])
+  end
+
+  def checkout
+    @books = @cart.books
+    @total_price = @books.sum(&:price)
+  end
+
+  def payment
+    begin
+      books = @cart.books
+      total_price = books.sum(&:price)
+
+      ActiveRecord::Base.transaction do
+        CartItem.where(cart_id: @cart.id).delete_all
+
+        @order = current_user.orders.new(
+          total_price: total_price,
+          status: :pending,
+          shipping_address: order_params[:shipping_address]
+        )
+
+        books.each do |book|
+          @order.order_books.new(
+            book: book,
+            book_title: book.title,
+            price_per_unit: book.price,
+            quantity: 1
+          )
+        end
+
+        @order.save!
+      end
+
+      payment_intent = Stripe::PaymentIntent.create(
+        amount: (total_price * 100).to_i,
+        currency: Constants::CURRENCY,
+        payment_method_types: Constants::STRIPE_PAYMENT_METHODS,
+        metadata: {
+          order_id: @order.id,
+          user_id: current_user.id,
+          book_ids: books.ids.join(",")
+        }
+      )
+      @client_secret = payment_intent.client_secret
+      @order.stripe_payment_intent = payment_intent.id
+      @order.save!
+    rescue Stripe::InvalidRequestError => e
+      err_message = handle_stripe_error_messages(e)
+      redirect_to order_checkout_path, alert: err_message
+    end
+  end
+
+  def complete
+    @order = current_user_orders.find_by(id: params[:id])
+  end
+
+  def reorder
+    old_order = current_user_orders.find_by(id: order_params[:id])
+    if old_order
+      begin
+        @order = old_order.deep_clone include: [:order_books], only: [:total_price, :shipping_address]
+        @order.user = current_user
+        @order.status = :pending
+        @order.save!
+
+        book_ids = @order.order_books.collect(&:book_id).join(",")
+        payment_intent = Stripe::PaymentIntent.create(
+          amount: (@order.total_price * 100).to_i,
+          currency: Constants::CURRENCY,
+          payment_method_types: Constants::STRIPE_PAYMENT_METHODS,
+          metadata: {
+            order_id: @order.id,
+            user_id: current_user.id,
+            book_ids: book_ids
```

Comment:
> Why do we need `book_ids` in metadata?

Currently, I think we don't need `book_ids` anymore. I will remove it. 

> If we need it, why don't we use an array instead of a string?

Because Stripe doesn't allow putting an array in metadata. I tried and it raised an error `Invalid value type: ... must be a string`